api/client/src/main/java/org/projectnessie/client/auth/oauth2/TokenExchangeConfig.java

```diff
@@ -71,19 +71,27 @@ static TokenExchangeConfig fromConfigSupplier(Function<String, String> config) {
     applyConfigOption(config, CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_SCOPES, builder::scope);
     applyConfigOption(config, CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_AUDIENCE, builder::audience);
     String subjectToken = config.apply(CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_SUBJECT_TOKEN);
+    String subjectTokenType = config.apply(CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_SUBJECT_TOKEN_TYPE);
     if (subjectToken != null) {
-      String subjectTokenType = config.apply(CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_SUBJECT_TOKEN_TYPE);
       builder.subjectToken(
           TypedToken.of(
               subjectToken,
               subjectTokenType == null ? URN_ACCESS_TOKEN : URI.create(subjectTokenType)));
+    } else if (subjectTokenType != null) {
+      builder.subjectTokenProvider(
+          (accessToken, refreshToken) ->
+              TypedToken.of(accessToken.getPayload(), URI.create(subjectTokenType)));
     }
     String actorToken = config.apply(CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_ACTOR_TOKEN);
+    String actorTokenType = config.apply(CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_ACTOR_TOKEN_TYPE);
     if (actorToken != null) {
-      String actorTokenType = config.apply(CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_ACTOR_TOKEN_TYPE);
       builder.actorToken(
           TypedToken.of(
               actorToken, actorTokenType == null ? URN_ACCESS_TOKEN : URI.create(actorTokenType)));
+    } else if (actorTokenType != null) {
+      builder.actorTokenProvider(
+          (accessToken, refreshToken) ->
+              TypedToken.of(accessToken.getPayload(), URI.create(actorTokenType)));
```

Comment:
nit: it might be best to do `URI.create()` when we get the config for any parsing errors to be detected earlier. WDYT?

Reply:
> I wonder if it might be best to have a different config for this

We are already adding something like a dozen new options, I'd be cautious adding another one. I think we can reuse the same option, but I will try to clarify the purpose in the javadocs. Would that work for you?

> it might be best to do URI.create() when we get the config

👍

Reply:
BTW I think what I did is not correct wrt to actor token types... the default is no actor token at all. I will fix that as well.

(I also plan to add an integration test for actor tokens later, but that may take some time to be done.)